components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -451,4 +453,43 @@ absl::optional<SolanaAddress> AssetDiscoveryManager::DecodeMintAddress(
   return SolanaAddress::FromBytes(pub_key_bytes);
 }
 
+// static
+// Creates a URL like
+// https://api.simplehash.com/api/v0/nfts/owners?chains={chains}&wallet_addresses={wallet_addresses}
+GURL AssetDiscoveryManager::GetSimpleHashNftsByWalletUrl(
+    const std::string& account_address,
+    const std::vector<std::string>& chain_ids) {
+  if (chain_ids.empty()) {
+    return GURL();
+  }
+
+  std::string url = base::StringPrintf("%s/api/v0/nfts/owners", kSimpleHashUrl);
+  const base::flat_map<std::string, std::string>& simple_hash_chain_ids =
+      SimpleHashChainIds();
+  std::string chain_ids_param;
+  for (const auto& chain_id : chain_ids) {
+    auto it = simple_hash_chain_ids.find(chain_id);
+    if (it != simple_hash_chain_ids.end()) {
+      if (!chain_ids_param.empty()) {
+        chain_ids_param += ",";
+      }
+      chain_ids_param += it->second;
+    }
+  }
+
+  if (chain_ids_param.empty()) {
+    return GURL();
+  }
+
+  // Add the chain IDs to the URL as a query parameter
+  url =
+      base::StringPrintf("%s?chains=%s", url.c_str(), chain_ids_param.c_str());
+
+  // Add the wallet address as a query parameter to the URL
+  url = base::StringPrintf("%s&wallet_addresses=%s", url.c_str(),
+                           account_address.c_str());
+
+  return GURL(url);
+}
```

Comment:
So I know we pref using `net::AppendQueryParameter` for this, but it doesn't support comma separated values - the commas get URL encoded.  And in this case, the URL parameters will not be user input.